LDEV-5871 Add cacheprefix attribute to query

deanmaunder opened the pull request:
https://luceeserver.atlassian.net/browse/LDEV-5871

michaeloffner committed:
release candidate

michaeloffner committed:
new cycle

cfmitrah committed:
For LDEV-5833, Added a test case for dateTimeFormat() with the masks 'l', 'll', and 'lll'.

zspitzer committed:
Merge pull request #2667 from MitrahSoft/LDEV-5833

For LDEV-5833, Added a test case for dateTimeFormat() with the masks 'l', 'll', and 'lll'.

michaeloffner committed:
LDEV-5842 - allow single digit hours and double white space

michaeloffner committed:
LDEV-5842 - allow single digit hours and double white space

michaeloffner committed:
LDEV-5659 - fix copy paste bug

michaeloffner committed:
LDEV-5856 - Update monitor installation to use new JSON config structure

michaeloffner committed:
LDEV-5863 - Add forced sync to config file writes to prevent Docker read race condition

michaeloffner committed:
LDEV-5863 - revert

zspitzer committed:
LDEV-5869 bugfix: cookie parsing should correctly handle values containing =

zspitzer committed:
LDEV-5873 bugfix: udfs.clear() in pageContent.release() needs additional check

deanmaunder committed:
LDEV-5871 Add cacheprefix attribute to query

deanmaunder committed:
LDEV-5871 Add cacheprefix attribute to query

zspitzer committed:
LDEV-296 throw an error when unsupported initMethod used

zspitzer committed:
LDEV-5878  GHA mirror dpocker images for CI to github container registry

zspitzer committed:
LDEV-5878 use ghcr.io github container registry mirror for all builds (6.2)

zspitzer commented:
@deanmaunder i fixed the error with CI, can you rebase please?

please rename the test without dash and use BDD style (describe/it/expect)

As we have multiple cache providers, like redis, memcached and ehcache, we need to test them all

https://github.com/lucee/Lucee/blob/7.0/test/_setupTestServices.cfc#L244

i.e. https://github.com/lucee/Lucee/blob/7.0/test/tickets/LDEV4342.cfc

see https://github.com/lucee/Lucee/blob/7.0/test/tags/dbInfo.cfc for such an approach

Also the tests don't yet cover `cacheRemove()` or `cacheClear()` which is quite an core important aspect for this nice new enhancement

zspitzer commented on test/tickets/LDEV-5871.cfc:
```diff
@@ -0,0 +1,345 @@
+		assertEquals(result1.executed, result2.executed, "Query with params should be cached");
+	}
+
+	private function repeatString(required string str, required numeric times) {
```
use the BIF ? https://docs.lucee.org/reference/functions/repeatstring.html

deanmaunder committed:
LDEV-5871 Update testing

deanmaunder committed:
LDEV-5871 Update testing

deanmaunder committed:
LDEV-5871 Update testing

deanmaunder committed:
LDEV-5871 Update testing

zspitzer committed:
fix bad test

deanmaunder committed:
LDEV-5871 Update tests to work with multiple caches

deanmaunder committed:
LDEV-5871 Update tests to work with multiple caches

deanmaunder committed:
LDEV-5871 Add test to remove specific path

deanmaunder committed:
LDEV-5871 Add test to remove specific path

deanmaunder committed:
LDEV-5871 Tests passing locally....

deanmaunder committed:
LDEV-5871 Tests passing locally....

zspitzer commented:
AFAIK This is still currently O(n) without additional changes?

deanmaunder committed:
LDEV-5871 Skip some memcached tests

deanmaunder committed:
LDEV-5871 Skip some memcached tests

deanmaunder commented:
> AFAIK This is still currently O(n) without additional changes?

Unsure what you are asking about?

michaeloffner committed:
add
          - jtds-extension

michaeloffner committed:
LDEV-5891 - check if connection is null

zspitzer committed:
LDEV-5063 test case

zspitzer committed:
LDEV-5903 testcase (disabled)

zspitzer committed:
remove accidental commit

michaeloffner committed:
LDEV-5903 - remove PhysicalClassLoader when closing Mapping

michaeloffner committed:
LDEV-5903 - clean up PhysicalClassLoader, move static methods to factory class and create internal Classloader

michaeloffner committed:
LDEV-5903 - move static methods to factory

deanmaunder commented:
@zspitzer anything further need to be done on this?

michaeloffner committed:
clean up

michaeloffner committed:
LDEV-5903 - replace PhysicalClassloader when reached a certain threshold

michaeloffner committed:
LDEV-5903 - replace PhysicalClassloader when reached a certain threshold

michaeloffner committed:
LDEV-5903 - use reference in staticScope

michaeloffner committed:
enable test case

michaeloffner committed:
enable test case

michaeloffner committed:
extend test case

michaeloffner committed:
improve test case

zspitzer committed:
LDEV-5931 fix local file access security in single mode

michaeloffner committed:
add test case for LDEV-5913

michaeloffner committed:
Merge branch '6.2' of https://github.com/lucee/Lucee into 6.2

michaeloffner committed:
extend test case for LDEV-5913 to show the unerlaying issue also exist in Lucee 6

zspitzer committed:
LDEV-5932 fix cfhttp compression support with GET

zspitzer committed:
bump tomcat to 9.0.112

zspitzer committed:
LDEV-5939 bugfix: validate directory is specified for delete/rename

zspitzer committed:
LDEV-5903 cleanup test to use sub folder

zspitzer committed:
LDEV-5942 sessionRotate() should rotate JSESSIONID for J2EE sessions

zspitzer committed:
LDEV-5730  fix JSession casting bug

zspitzer committed:
LDEV-3248 testcase

zspitzer committed:
LDEV-5942 gracefully handle fake jee sessions under jsr-223

zspitzer committed:
Merge pull request #2717 from zspitzer/LDEV-5942-v6

LDEV-5942 backport jee session fixes

zspitzer committed:
set version

zspitzer committed:
cleanup

zspitzer committed:
LDEV-5943 fix inspect auto defaults (6.2)

zspitzer committed:
LDEV-5943 fix inspect template fallback when not configured

zspitzer committed:
LDEV-5884 fix version comparison copy paste bug (6.2)

michaeloffner committed:
LDEV-5930 - improve change check in storage scopes.

zspitzer committed:
LDEV-5954 query.addParam should support arrays when list=true (6.2)

michaeloffner committed:
release candidate

michaeloffner committed:
new release cycle

zspitzer committed:
adding warn logging to _removeContext()

zspitzer committed:
LDEV-5962 fix pool config: maxTotal default 0, maxWaitMillis 30s

zspitzer committed:
set version

zspitzer committed:
Merge branch '6.2' of https://github.com/lucee/Lucee into 6.2

zspitzer committed:
LDEV-5964 reuse connection in store() to prevent nested borrow deadlock

zspitzer committed:
set version

zspitzer committed:
LDEV-5963 wire idleTimeout to minEvictableIdleTimeMillis

zspitzer committed:
set version

zspitzer committed:
cleanup

zspitzer committed:
LDEV-5965 return false when datasource connection validation throws exception

zspitzer committed:
LDEV-5966 return connection to pool on timeout instead of closing directly (#2727)

zspitzer committed:
set version

zspitzer committed:
LDEV-5972 refactor modern mssql approach (6.2)

zspitzer committed:
LDEV-5976  Error merging cfconfig.json

zspitzer committed:
LDEV-5961 disable slow pool connection tests

zspitzer committed:
LDEV-5991 change default datasource idle timeout to 10m

zspitzer committed:
LDEV-5991 update the other driver.cfc to connectionTimeout 10m

michaeloffner committed:
LDEV-6020 - order arguments keys for key creation

michaeloffner committed:
LDEV-6021 - check application context for default caches

michaeloffner committed:
add logging to PhysicalClassloader to get inside about classloading thresholds

michaeloffner committed:
LDEV-6026 - add possibility to disable log entry

michaeloffner committed:
LDEV-5903 - enable optional class updating and change thresholds for classloader flushing

michaeloffner committed:
LDEV-5903 - enable optional class updating and change thresholds for classloader flushing

michaeloffner committed:
LDEV-5903 - enable optional class updating and change thresholds for classloader flushing

zspitzer committed:
LDEV-6043 fix DecimalFormat bug

zspitzer committed:
LDEV-6046 sessionInvalidate() de-activate old session when using database storage

zspitzer committed:
LDEV-6048 fix loader LUCEE_VERSION warning check

zspitzer committed:
LDEV-6051 default datasource maxIdle to connectionLimit (6.2)

https://luceeserver.atlassian.net/browse/LDEV-6051

zspitzer committed:
Merge pull request #2741 from zspitzer/LDEV-6051-maxIdle-v6

LDEV-6051 default datasource maxIdle to connectionLimit (6.2)

zspitzer committed:
set version

michaeloffner committed:
remove provider no longer supported

michaeloffner committed:
remove provider no longer supported

michaeloffner committed:
LDEV-6057 - handle null

zspitzer committed:
LDEV-6063 cfflush should not set Connection: close

zspitzer committed:
LDEV-6065 force null support=false for the admin

zspitzer committed:
LDEV-6067 add admin updates notification test

zspitzer committed:
LDEV-6066 streamline verifyDatasource

zspitzer committed:
LDEV-6068 refactor admin update version handling

zspitzer committed:
LDEV-6069 fix admin ext version comparison

zspitzer committed:
LDEV-6072 test case for bugs due to missing pageSource hashcode()

zspitzer committed:
LDEV-6056 fix pageSourceImpl equals check

zspitzer committed:
LDEV-6072 add pageSourceImpl hashcode override

zspitzer committed:
tests, reduce noise from windows locking with createObject (backport from 7)

michaeloffner committed:
Merge pull request #2750 from zspitzer/LDEV-6072-fix-62

LDEV-6072 LDEV-6056  add pageSourceImpl hashcode override and fix pageSourceImpl equals check

michaeloffner committed:
LDEV-6056 - check mapping as well

michaeloffner committed:
LDEV-6056 - check mapping as well

michaeloffner committed:
LDEV-6056 - fix regression caused by invalid equal

michaeloffner committed:
remove unnecessary throws defintion

michaeloffner committed:
LDEV-5903 - fix PageSourcePool not flushed

Because there is no 1:1 relation between mappings and pools anymore, this code was no longer working as expected, we simply flush all pools having a reference to that classloader.
Maybe a bit to much, but this is an action that should happen not that much.
Also updating the function PagePoolClear to use the same functionality that properly flushes all Application mappings, not only the once from the current request.

michaeloffner committed:
LDEV-5903 - in ClazzDynamic use the identity hash code (memory reference) instead of the Classloader as mapping

michaeloffner committed:
LDEV-5903 - use correct hash code / make SoftReference for loaders /

michaeloffner committed:
LDEV-5903 - make sure PageSourcePoolWatcherThread stops faster and enforce it

zspitzer committed:
LDEV-5903 improve logging (detail and move some log to TRACE

zspitzer committed:
LDEV-6084 test case for IllegalAccessError

zspitzer committed:
LDEV-6088 add test to check and report ext bundle problems (v6)

michaeloffner committed:
LDEV-6097 - cache NativeException and create CatchBlock only once within a PageException

michaeloffner committed:
LDEV-6097 - cache NativeException and create CatchBlock only once within a PageException

michaeloffner committed:
update PDF extension

michaeloffner committed:
update Image extension

michaeloffner committed:
update ESAPI extension

michaeloffner committed:
release candidate

zspitzer committed:
LDEV-6114 testcase for multi-byte unicode chars break string functions

michaeloffner committed:
LDEV-6120 - look for javax specific tag class first

michaeloffner committed:
Merge branch '6.2' of https://github.com/lucee/Lucee into 6.2

michaeloffner committed:
LDEV-6120 - look for javax specific tag class first

michaeloffner committed:
update to newer redis version that now also support javax

michaeloffner committed:
update compress extension

michaeloffner committed:
update zip test case

michaeloffner committed:
revert compress extension

michaeloffner committed:
update chart extension used for testing

michaeloffner committed:
enable test case

michaeloffner committed:
update chart extension used for testing

michaeloffner committed:
keep cause

michaeloffner committed:
add new tags to dtd

michaeloffner committed:
ignore all tags inside group (only supported with Lucee 7, causing problems in Lucee 6 parser)

michaeloffner committed:
improve test cases

michaeloffner committed:
LDEV-6073 - add formtag-form to lucee-context.lar

michaeloffner committed:
RC

michaeloffner committed:
LDEV-5903 - log the creation of new PhysicalClassloaders

michaeloffner committed:
LDEV-5903 - add additional logging

zspitzer committed:
LDEV-5930 add disabled test for session component property loss

zspitzer committed:
Merge branch '6.2' of https://github.com/lucee/Lucee into 6.2

michaeloffner committed:
LDEV-5903 - add additional logging

michaeloffner committed:
Merge branch '6.2' of https://github.com/lucee/Lucee into 6.2

michaeloffner committed:
LDEV-6132 - use file checksums as JavaSettings cache key to optimize ClassLoader reuse

michaeloffner committed:
release candidate

michaeloffner committed:
test commit

michaeloffner committed:
test commit

michaeloffner committed:
test commit

michaeloffner committed:
enable node functionality again (needs to be updated until june)

michaeloffner committed:
enable node functionality again (needs to be updated until june)

michaeloffner committed:
LDEV-6133 - clean Classloader ideling for some time.

michaeloffner committed:
clean up

michaeloffner committed:
release

michaeloffner committed:
change version appendix (same version)

michaeloffner committed:
update version because extension have changed same version but with no appendix (RC)

zspitzer committed:
LDEV-6150 add HTTP connection pool stats to getSystemMetrics()

zspitzer committed:
bump tomcat to 9.0.115

zspitzer committed:
LDEV-6162 fix overwriting of graph.cfm

zspitzer committed:
LDEV-6086 InternalRequest swallows RequestTimeoutException exceptions

zspitzer committed:
LDEV-6086 avoid killing the parent request

zspitzer committed:
LDEV-6165 skip OSGi Export-Package validation for LAR archives

zspitzer committed:
LDEV-6165 cleanup

zspitzer committed:
LDEV-6172 LDEV-6173 add test coverage for Application.cfc archive mappings

zspitzer committed:
LDEV-6172 fix physicalFirst override in MappingImpl lazy init

zspitzer committed:
LDEV-6177 prevent temp directory duplication in security file access list

zspitzer committed:
LDEV-6173 fix custom tag archive test filename convention

zspitzer committed:
LDEV-6133 reduce log levels for physicalClassLoader to TRACE

zspitzer committed:
LDEV-6168 Remove static NativeException LRUMap cache

zspitzer committed:
cleanup tests

cfmitrah committed:
For LDEV-6185, added a test case for null support in serializeJSON()

zspitzer committed:
Merge pull request #2771 from MitrahSoft/LDEV-6185

For LDEV-6185, added a test case for null support in serializeJSON()

zspitzer committed:
LDEV-6188 add test case for optional arguments scope with nullSupport=true

zspitzer committed:
LDEV-6190 fix classloader factory returning stale classloader after flush

zspitzer committed:
LDEV-6196 add contextPassageLength to AddionalAttrs thread-local

deanmaunder commented:
Sorry @zspitzer this one has become critical in our environment, can we get it merged?  Will it backport to 6?  we arent ready for 7 yet.
Otherwise I can build a custom Lucee version and deploy.

michaeloffner committed:
LDEV-6190 - remove close method call from flush

michaeloffner committed:
LDEV-6171 - make generated code more independent of jakarta/javax packages (not a 100%)

michaeloffner committed:
LDEV-6171 - make generated code more independent of jakarta/javax packages (not a 100%)

michaeloffner committed:
LDEV-6123 - floor to long instead of int, fixing silent truncation of large numbers outside 32-bit range

michaeloffner committed:
LDEV-6144 - patch for jar name conflict

michaeloffner committed:
LDEV-6144 - start felix in case of an exception

michaeloffner committed:
LDEV-6135 - fix doJoin() timeout calculation draining against wall clock

Previously _timeout was decremented by total elapsed time since the start
of the join loop, including overhead from iterating already-completed
threads. This caused the timeout to expire prematurely — e.g. a 60s
timeout would be exhausted after ~1.5s when iterating 44 threads.

Fix: recalculate remaining time fresh each iteration as
(start + timeout) - currentTimeMillis(), so only real elapsed wall time
is counted, not accumulated loop overhead.

michaeloffner committed:
LDEV-6201 - add class name to the key

michaeloffner committed:
LDEV-6121- fix comparsion

michaeloffner committed:
set version

deanmaunder committed:
Merge origin/6.2 into LDEV-5871

deanmaunder commented:
> AFAIK This is still currently O(n) without additional changes?

You're right that key iteration is still O(n) with the default CacheSupport.remove(CacheKeyFilter) impl. What this PR delivers is skipping value deserialization on cacheClear:

- cacheClear(tags, "query") → QueryTagFilter (CacheEntryFilter) → ce.getValue() on every entry → full deserialization
- cacheClear("prefix:*", "query") → WildCardFilter (CacheKeyFilter) → keys-only scan, no value reads

The issue we were trying to solve was the storm that happened during a cacheClear on Redis where it would read the value of hundreds of large query objects from the cache, just to get the tags.
